tests/test_evaluation.py

```diff
@@ -158,6 +158,13 @@ def test_operator_in():
     assert isinstance(operator_in(celtypes.IntType(-1), container_2), CELEvalError)
 
 
+def test_function_matches():
```

Comment:
This seems to assume that RE2 is installed as part of the testing. I'm not sure that's best. I may be wrong about this, but I can imagine someone declining the RE2 add-on and then being unable to test CEL.

Maybe there should be a `@pytest.skipif()` option to test the two low-level functions depending on the `_USE_RE2` global,  as well as a separate generic test of `function_matches()`? 

(I love the test of an RE2-specific feature, that's fabulous!)

Reply:
Anyone testing `cel-python` will have to install `google-re2-stubs` if they want to type-check the code and ...

`google-re2-stubs` depends on `google-re2` because that is the way [1] for it to indicate which versions of `google-re2` it is compatible with and also that it is intended to be a stub for the `google-re2` distribution of `re2` and not one of the other distributions of `re2` which all claim to be "drop-in" replacements for `re` but in practice have slightly different APIs.

So, if we need to type-check we need the stubs and installing the stubs installs the implementation, and we cannot conditionally install the stubs because the code will not type-check unless we install the stubs.

I happen to be the maintainer of `google-re2-stubs` so I can obviously drop the dependency on `google-re2` so that `cel-python` can install the stubs without installing `google-re2` but given the weirdness in the `re2` ecosystem noted above, I think that is not optimal for other reasons.

# References

[1]

> In addition, stub-only distributions MAY indicate which version(s) of the runtime package are targeted by indicating the runtime distribution’s version(s) through normal dependency data. For example, the stub package flyingcircus-stubs can indicate the versions of the runtime flyingcircus distribution it supports through dependencies field in pyproject.toml.

https://typing.readthedocs.io/en/latest/spec/distributing.html#stub-only-packages